core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -834,24 +803,24 @@ oppia.directive('conversationSkin', [
                     }
                     FatigueDetectionService.reset();
                     NumberAttemptsService.reset();
-                    _nextFocusLabel = FocusManagerService.generateFocusLabel();
+                    _nextFocusLabel = focusLabel;
 
-                    PlayerTranscriptService.setDestination(newStateName);
+                    PlayerTranscriptService.setDestination(
+                      nextCard.getStateName());
 
                     // These are used to compute the dimensions for the
                     // next card.
-                    $scope.upcomingStateName = newStateName;
-                    $scope.upcomingParams = newParams;
-                    $scope.upcomingContentHtml = (
-                      contentHtml + _getRandomSuffix());
+                    $scope.nextCard = nextCard;
+                    $scope.nextCard.addRandomSuffixToContentHtml(
```

Comment:
All this random suffix stuff should _probably_ be generated (by the exploration engine service?) before the state service passes the card to the conversation skin directive.

Reply:
I think it was like that before, but _getRandomSuffix was a general function and had nothing to do with an exploration or pretest. So, I think this was moved to conversation skin where this could be done for both generally.

Reply:
Ah I see. No I think it should be hidden within the engines. It's fine to duplicate this one though, or maybe we can make a general-purpose RandomStringService.

Reply:
Done.